### PR TITLE
Actually set HTTP version in response

### PR DESCRIPTION
**Pull request by otoolep**
The HTTP version was being set using information in the Server, but the Server's version wasn't set at the time. Move version into `NewServer` so it's now required.

Version is worth special-casing here, since the build information is pretty important, and many components need it.

### CODE REVIEW

**Comment by otoolep** [comment]
@neonstalwart @corylanou

**Comment by otoolep** [comment]
````
go build -ldflags="-X main.version v1234 -X main.commit xxx"
.
.
.
$ curl -v -G http://localhost:8086/query --data-urlencode "q=CREATE DATABASE foo"
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8086 (#0)
> GET /query?q=CREATE%20DATABASE%20foo HTTP/1.1
> User-Agent: curl/7.35.0
> Host: localhost:8086
> Accept: */*
> 
< HTTP/1.1 200 OK
< Content-Type: application/json
< Request-Id: 21222deb-1133-11e5-8001-000000000000
< X-Influxdb-Version: v1234
< Date: Fri, 12 Jun 2015 18:45:00 GMT
< Content-Length: 49
< 
* Connection #0 to host localhost left intact
{"results":[{"error":"database already exists"}]}
```

**Comment by corylanou** [comment]
@otoolep Lets actually add a test this time :-).

**Comment by otoolep** [comment]
Unit test added @corylanou

**Comment by corylanou** [comment]
+1

File: cmd/influxd/run/server_test.go [hunk 1]
```diff
@@ -10,6 +11,19 @@ import (
...
 
+// Ensure that HTTP responses include the InfluxDB version.
+func TestServer_HTTPResponseVersion(t *testing.T) {
+	version := "v1234"
+	s := OpenServerWithVersion(NewConfig(), version)
+	defer s.Close()
+
+	resp, _ := http.Get(s.URL() + "/query")
+	got := resp.Header.Get("X-Influxdb-Version")
+	if version != version {
```
Comment (neonstalwart) [inline cmd/influxd/run/server_test.go]: this test will never fail :wink: 

`if got != version`

**Comment by neonstalwart** [comment]
thanks for the fix @otoolep - my PR was a drive-by fix, sorry about that.  since now you've looked at this same code, i was wondering what you thought about putting the value for the X-Influxdb-Version header in the config?  by default it could be based on the build version but i think it could also be a feature that the user could set it themselves to some arbitrary value.  each of the constructors that needed the version already had the config and it would make it simpler.

**Comment by otoolep** [comment]
I thought about the config @neonstalwart but decided to leave the config just for actual config stuff.

**Comment by neonstalwart** [comment]
fair enough - that's the way i leaned too.

**Pull request merged**